### PR TITLE
Enable AIP-44 and AIP-52 by default for development and CI on main

### PR DESCRIPTION
**Pull request by potiuk**
The AIP-44 and AIP-52 are controlled now by environment variables, however those variables were not passed by default to inside the docker-compose environment so they had no effect when set on the ci.yaml. This PR fixes it, but it also sets the variables to be enabled by default in Breeze environment and when the tests are run locally in main using local venv so that the contributors are not surprised when they try to reproduce local failures.

In 2.6 branch, we will set both variables to "false" by default in ci.yml, so that the tests are not run when we cherry-pick the changes.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -38,6 +38,8 @@
 os.environ["AIRFLOW__CORE__UNIT_TEST_MODE"] = "True"
 os.environ["AWS_DEFAULT_REGION"] = os.environ.get("AWS_DEFAULT_REGION") or "us-east-1"
 os.environ["CREDENTIALS_DIR"] = os.environ.get("CREDENTIALS_DIR") or "/files/airflow-breeze-config/keys"
+os.environ["AIRFLOW_ENABLE_AIP_44"] = os.environ.get("AIRFLOW_ENABLE_AIP_44") or "true"
```
Comment (potiuk) [inline tests/conftest.py:41]: This is the default if not set when our tests are run. This way the user wil not have to set the variables in their tests when they run in from the IDE (cc: @ephraimbuddy @jedcunningham @eladkal @pierrejeambrun ). I found it highly annoying that I had to set it in main - but still in airflow they will be disabled by default.

In CI in branch 2.6 we will set the vars to be "false" by default, so the tests will be skipped there - even if the change here will be part of 2.6.

File: scripts/ci/docker-compose/devcontainer.env [hunk 2]
```diff
@@ -17,6 +17,8 @@
 HOME=
 AIRFLOW_CI_IMAGE="ghcr.io/apache/airflow/main/ci/python3.7:latest"
 ANSWER=
+AIRFLOW_ENABLE_AIP_44="true"
+AIRFLOW_ENABLE_AIP_44="true"
```
Comment (ephraimbuddy) [inline scripts/ci/docker-compose/devcontainer.env]: Should one be AIP 52 here?

Comment (potiuk) [inline scripts/ci/docker-compose/devcontainer.env]:
```suggestion
AIRFLOW_ENABLE_AIP_52="true"
```

Comment (potiuk) [reply scripts/ci/docker-compose/devcontainer.env]: Very much so :)

**Review by ephraimbuddy — APPROVED** [review]

**Pull request merged**